File: FrontEndIndividual/app/src/main/java/com/example/frontendindividual/dto/Usuario.kt
package com.example.frontendindividual.dto

import java.util.*
import kotlin.collections.ArrayList

class Usuario(
    var id: Int,
    var createdAt: Long,
    var updatedAt: Long,
    var nombre: String,
    var apellido:String,
    var cedula:String,
    var fechaDeNacimiento:String,
    var perfil:String,
    var contrasenia:String,
    var mascotasDeUsuario:ArrayList<Mascota>,
    var rolesDeUsuario:ArrayList<RolUsuario>
) {
    var fechaCreacion: Date
    var fechaActualizacion: Date

    init {
        fechaCreacion = Date(createdAt)
        fechaActualizacion = Date(updatedAt)
    }


}

File: backend-individual/backend-individual/api/models/Usuario.js
/**
 * Usuario.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
    nombre:{
      type: 'string'
    },
    apellido:{
      type: 'string'
    },
    email:{
    type:'string'
    },
    cedula:{
      type: 'string'
    },
    fechaDeNacimiento:{
      type: 'string'
    },
    perfil:{
      type:'string'
    },
    contrasenia:{
      type:'string'
    },
    mascotasDeUsuario:{
      collection:'mascota',
      via: 'fkUsuario'
    },
    rolesDeUsuario: {
      collection: 'rolUsuario',
      via: 'fkUsuario'
    }
  },

};

